crates/api_common/src/utils.rs

```diff
@@ -60,6 +61,18 @@ pub async fn is_mod_or_admin(
   Ok(())
 }
 
+pub async fn is_top_admin(pool: &DbPool, person_id: PersonId) -> Result<(), LemmyError> {
+  let admins = PersonViewSafe::admins(pool).await?;
+  let top_admin = admins
+    .get(0)
+    .ok_or_else(|| LemmyError::from_message("no admins"))?;
```

Comment:
This is a nice way to convert a vector to an error if that index is missing. No idea why rust lets you get slices and them not be options / results, potentially throwing runtime errors.